### PR TITLE
feat(driver): check `maxBytesPerTxList` for compressed txlist bytes

### PR DESCRIPTION
**Pull request by davidtaikocha**

### CODE REVIEW

File: pkg/txlist_validator/tx_list_validator.go [hunk 1]
```diff
@@ -29,24 +30,38 @@ func NewTxListValidator(
...
+		txs types.Transactions
+		err error
+	)
+
+	if txListBytes, err = utils.Decompress(txListBytes); err != nil {
+		log.Info("Failed to decompress tx list bytes", "blockID", blockID, "error", err)
+		return false
+	}
+
+	if err = rlp.DecodeBytes(txListBytes, &txs); err != nil {
```
Comment (smtmfft) [inline pkg/txlist_validator/tx_list_validator.go:64]: here decompress & decode are only tentative operation?? Can it return final txs to save dup decompress & decode later on (I guess it's there, not sure).

**Review by smtmfft — APPROVED** [review]

File: driver/txlist_decompressor/txlist_decomporessor.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+// TryDecomporess validates and decompresses whether the transactions list in the TaikoL1.proposeBlock transaction's
+// input data is valid, the rules are:
+// - If the transaction list is empty, it's valid.
+// - If the transaction list is not empty:
+//  1. If the transaction list is using calldata, the compressed bytes of the transaction list must be
+//     less than or equal to maxBytesPerTxList.
+//  2. The transaction list bytes must be able to be RLP decoded into a list of transactions.
+func (v *TxListDecompressor) TryDecomporess(
```
Comment (RogerLamTd) [inline driver/txlist_decompressor/txlist_decomporessor.go:40]:
```suggestion
func (v *TxListDecompressor) TryDecompress(
```

File: driver/chain_syncer/blob/syncer.go [hunk 3]
```diff
@@ -72,7 +72,7 @@ func NewSyncer(
 		state:             state,
 		progressTracker:   progressTracker,
 		anchorConstructor: constructor,
-		txListValidator: txListValidator.NewTxListValidator(
+		txListDecompressor: txListDecomporessor.NewTxListDecompressor(
```
Comment (RogerLamTd) [inline driver/chain_syncer/blob/syncer.go:75]:
```suggestion
		txListDecompressor: txListDecompressor.NewTxListDecompressor(
```

File: driver/chain_syncer/blob/syncer.go [hunk 4]
```diff
@@ -246,38 +246,29 @@ func (s *Syncer) onBlockProposed(
 	}
 
 	// Decode transactions list.
-	var txListDecoder txlistfetcher.TxListFetcher
+	var txListFecher txlistFetcher.TxListFetcher
```
Comment (RogerLamTd) [inline driver/chain_syncer/blob/syncer.go:249]:
```suggestion
	var txListFetcher txlistFetcher.TxListFetcher
```

File: driver/chain_syncer/blob/syncer.go [hunk 5]
```diff
@@ -246,38 +246,29 @@ func (s *Syncer) onBlockProposed(
...
 	if event.Meta.BlobUsed {
-		txListDecoder = txlistfetcher.NewBlobTxListFetcher(s.rpc.L1Beacon, s.blobDatasource)
+		txListFecher = txlistFetcher.NewBlobTxListFetcher(s.rpc.L1Beacon, s.blobDatasource)
```
Comment (RogerLamTd) [inline driver/chain_syncer/blob/syncer.go:251]:
```suggestion
		txListFetcher = txlistFetcher.NewBlobTxListFetcher(s.rpc.L1Beacon, s.blobDatasource)
```

File: driver/chain_syncer/blob/syncer.go [hunk 6]
```diff
@@ -246,38 +246,29 @@ func (s *Syncer) onBlockProposed(
...
 	} else {
-		txListDecoder = new(txlistfetcher.CalldataFetcher)
+		txListFecher = new(txlistFetcher.CalldataFetcher)
```
Comment (RogerLamTd) [inline driver/chain_syncer/blob/syncer.go:253]:
```suggestion
		txListFetcher = new(txlistFetcher.CalldataFetcher)
```

File: driver/chain_syncer/blob/syncer.go [hunk 7]
```diff
@@ -246,38 +246,29 @@ func (s *Syncer) onBlockProposed(
...
 	}
-	txListBytes, err := txListDecoder.Fetch(ctx, tx, &event.Meta)
+	txListBytes, err := txListFecher.Fetch(ctx, tx, &event.Meta)
```
Comment (RogerLamTd) [inline driver/chain_syncer/blob/syncer.go:255]:
```suggestion
	txListBytes, err := txListFetcher.Fetch(ctx, tx, &event.Meta)
```

File: driver/chain_syncer/blob/syncer.go [hunk 8]
```diff
@@ -246,38 +246,29 @@ func (s *Syncer) onBlockProposed(
...
-	}
-
+	// Decompress the transactions list and try to insert a new head block to L2 EE.
 	payloadData, err := s.insertNewHead(
 		ctx,
 		event,
 		parent,
 		s.state.GetHeadBlockID(),
-		txListBytes,
+		s.txListDecompressor.TryDecomporess(event.BlockId, txListBytes, event.Meta.BlobUsed),
```
Comment (RogerLamTd) [inline driver/chain_syncer/blob/syncer.go:271]:
```suggestion
		s.txListDecompressor.TryDecompress(event.BlockId, txListBytes, event.Meta.BlobUsed),
```

File: driver/txlist_decompressor/txlist_decomporessor_test.go [hunk 9]
```diff
@@ -30,48 +31,50 @@ var (
 	}
 )
 
-func TestIsTxListValid(t *testing.T) {
-	v := NewTxListValidator(
+func TestDecomporess(t *testing.T) {
```
Comment (RogerLamTd) [inline driver/txlist_decompressor/txlist_decomporessor_test.go:34]:
```suggestion
func TestDecompress(t *testing.T) {
```

File: driver/txlist_decompressor/txlist_decomporessor_test.go [hunk 10]
```diff
@@ -30,48 +31,50 @@ var (
...
 			rlpEncodedTransactionBytes(1, true),
-			true,
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			isValid := v.ValidateTxList(tt.blockID, tt.txListBytes, false)
-			require.Equal(t, tt.isValid, isValid)
+			require.Equal(t, tt.decompressed, d.TryDecomporess(tt.blockID, tt.txListBytes, false))
```
Comment (RogerLamTd) [inline driver/txlist_decompressor/txlist_decomporessor_test.go:77]:
```suggestion
			require.Equal(t, tt.decompressed, d.TryDecompress(tt.blockID, tt.txListBytes, false))
```

**Pull request merged**